### PR TITLE
Prune deprecated: classification and regression

### PR DESCRIPTION
**Pull request by ashutoshml**
## What does this PR do?

Fixes part of #771 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ashutoshml** [comment]
Failing integration test for `python 3.6`. Not sure how to resolve this.

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Removed deprecated functions, and warnings in Text ([#773](https://github.com/PyTorchLightning/metrics/pull/773))
   * `functional.wer`
   * `WER`
+- Removed deprecated functions, and warnings in Text ([#806](https://github.com/PyTorchLightning/metrics/pull/806))
```
Comment (Borda) [inline CHANGELOG.md:28]:
```suggestion
- Removed deprecated functions, and warnings in classification and regression ([#806](https://github.com/PyTorchLightning/metrics/pull/806))
```

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki seems we have the same issue with PL compatibility here too, as `F1` was one of the very legacy metrics first implemented in PL itself... :rabbit: 

so shall we break the compatibility with PL `v1.3` or preserve this burden till TM `v1.0`?
cc: @PyTorchLightning/core-metrics @PyTorchLightning/core-lightning

**Comment by SkafteNicki** [comment]
@Borda this is a though one, but I am in favour of breaking backwards-compatibility with lightning. We need first and foremost to think of metrics as its own package which does not depend on lightning. In practise I do not know how many actually uses metrics with base torch and how many uses it with lightning.
However, I would also be more comfortable with this breaking change if lightning v1.6 is out before v0.8 of metrics. This would mean that TM v0.8 is compatible with two major versions of lightning v1.5 and v1.6. If you are using lightning <v1.5 we recommend sticking with TM v0.7.

**Comment by Borda** [comment]
> However, I would also be more comfortable with this breaking change if lightning v1.6 is out before v0.8 of metrics. This would mean that TM v0.8 is compatible with two major versions of lightning v1.5 and v1.6. If you are using lightning <v1.5 we recommend sticking with TM v0.7.

we have this testing already inside EcoCI - https://github.com/PyTorchLightning/ecosystem-ci/blob/main/configs/PyTorchLightning/metrics_pl-develop.yaml

**Comment by awaelchli** [comment]
> However, I would also be more comfortable with this breaking change if lightning v1.6 is out before v0.8 of metrics. This would mean that TM v0.8 is compatible with two major versions of lightning v1.5 and v1.6. If you are using lightning <v1.5 we recommend sticking with TM v0.7.

+1 for this approach

**Comment by ashutoshml** [comment]
@Borda: Any updates on the final approach for this issue?

**Comment by Borda** [comment]
> @Borda: Any updates on the final approach for this issue?

it is going to merge as soon as all test passes :)

**Pull request merged**